### PR TITLE
feat/addMusic-system

### PR DESCRIPTION
**Pull request by Dachikkinho**

### CODE REVIEW

File: app/components/addMusicSubComponents/MusicForm.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+    useEffect(() => {
+        if (id) {
+            axios
+                .get(`https://mukhambazi-back.onrender.com/music/${id}`)
+                .then((res) => {
+                    reset(res.data);
+                })
+                .catch((err) => {
+                    console.log(err);
```
Comment (lursmana1) [inline app/components/addMusicSubComponents/MusicForm.tsx:36]: alert

File: app/components/addMusicSubComponents/MusicForm.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                });
+        }
+    }, [id]);
+
+    useEffect(() => {
+        axios.get('https://mukhambazi-back.onrender.com/album/').then((res) => {
+            setAlbums(res.data);
+        });
+
+        axios.get('https://mukhambazi-back.onrender.com/authors/').then((res) => {
```
Comment (lursmana1) [inline app/components/addMusicSubComponents/MusicForm.tsx]: / on the end is this correct ?

File: app/components/addMusicSubComponents/MusicForm.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            setArtists(res.data);
+        });
+    }, []);
```
Comment (lursmana1) [inline app/components/addMusicSubComponents/MusicForm.tsx:51]: dependency is needed i think

File: app/components/addMusicSubComponents/MusicForm.tsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+                            'Content-Type': 'multipart/form-data',
+                        },
+                    },
+                );
+            }
+            setUploaded(true);
+            setUploadedName(music.name);
+            reset();
+        } catch (err) {
+            console.log(err);
```
Comment (lursmana1) [inline app/components/addMusicSubComponents/MusicForm.tsx:92]: alert

File: app/components/addMusicSubComponents/MusicForm.tsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+                        image: music.image[0],
+                    },
+                    {
+                        headers: {
+                            'Content-Type': 'multipart/form-data',
+                        },
+                    },
+                );
+            }
+            setUploaded(true);
```
Comment (lursmana1) [inline app/components/addMusicSubComponents/MusicForm.tsx:88]: make this in then

File: app/components/addMusicSubComponents/MusicForm.tsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+import { CreateMusic } from '@/app/interfaces/createMusic.interface';
+import { ErrorMessage } from '@/app/components/ErrorMessage/ErrorMessage';
+import { Done } from '@/app/components/Done/Done';
+import styles from '../../(authorised)/addMusic/page.module.scss';
+import AdminSelect from '../MusicSelect/MusicSelect';
+import { Album } from '@/app/interfaces/album.interface';
+import { Artists } from '@/app/interfaces/createArtist.interface';
+import MusicSelect from '../MusicSelect/MusicSelect';
+
+export default function MusicForm() {
```
Comment (lursmana1) [inline app/components/addMusicSubComponents/MusicForm.tsx:14]: also Giant component split up

File: app/components/addMusicSubComponents/MusicForm.tsx (hunk 2 above)
Comment (Dachikkinho) [reply app/components/addMusicSubComponents/MusicForm.tsx]: yes it is.

File: app/components/addMusicSubComponents/MusicForm.tsx (hunk 3 above)
Comment (Dachikkinho) [reply app/components/addMusicSubComponents/MusicForm.tsx:51]: everything work correctly.

File: app/components/addMusicSubComponents/MusicFormStatus.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import styles from '../../(authorised)/addMusic/page.module.scss';
+import { Done } from '@/app/components/Done/Done';
+
+type Props = {
+    id: string;
```
Comment (lursmana1) [inline app/components/addMusicSubComponents/MusicFormStatus.tsx]: id string ?

**Pull request merged**